doc/download.xml

```diff
@@ -0,0 +1,80 @@
+<!-- ------------------------------------------------------------------- -->
+<!--                                                                     -->
+<!--  download.xml            Utils documentation                        -->
+<!--                                                                     -->
+<!--  Copyright (C) 2022, The GAP Group                                  --> 
+<!--                                                                     -->
+<!-- ------------------------------------------------------------------- -->
+
+<?xml version="1.0" encoding="UTF-8"?>
+
+<Chapter Label="chap-download">
+<Heading>Web Downloads</Heading>
+
+<Section Label="sec-download">
+<Heading>Functions for downloading files from the web</Heading>
+
+<ManSection>
+<Func Name="Download" Arg="url[, opt]" />
+
+<Description>
+This function downloads the file with the web address <A>url</A>,
+which must be a string.
+<P/>
+The result is a record which has at least the component <C>success</C>,
+with value <K>true</K> if the download was successful and <K>false</K>
+otherwise.
+In the former case, the component <C>result</C> is bound,
+whose value is a string that contains the contents of the downloaded file.
+In the latter case, the component <C>error</C> is bound,
+whose value is a string that describes the problem.
+<P/>
+The function calls the methods stored in the global list
+<C>Download_Methods</C> until one of them is successful.
+Currently there are methods based on the &GAP; functions
+<Ref Func="DownloadURL" BookName="curl"/> and
+<Ref Func="SingleHTTPRequest" BookName="io"/>,
+and methods based on the external programs <C>wget</C> and <C>curl</C>.
+<P/>
+An optional record <A>opt</A> can be given.
+The following components are supported.
+<P/>
+<List>
+<Mark><C>target</C></Mark>
+<Item>
+  If this component is bound then its value must be a string
+  that is a local filename,
+  and the function writes the downloaded contents to this file;
+  the returned record does not have a <C>result</C> component in this case.
+</Item>
+<Mark><C>verifyCert</C></Mark>
+<Item>
+  If this component is bound and has the value <K>false</K>
+  then those download methods that are based on <C>curl</C> will
+  omit the check of the server's certificate,
+  by calling <C>curl</C> with the <C>-k</C> option or setting the
+  <C>libcurl</C> option <C>CURLOPT_SSL_VERIFYPEER</C> to <C>0</C>.
```

Comment:
Should we really mention / document these low-level details? Most users won't care; those who do probably are savvy enough to look it up in the implementation?

Note that this explanation will likely be incomplete as soon as additional download methods are added.